website_sale_charge_payment_fee/static/src/js/website_sale_fee.js

```diff
@@ -1,29 +1,23 @@
-$(document).ready(function() {
-    if (_.str.startsWith(window.location.pathname, "/shop/payment")) {
-        var $pay_button = $(".oe_sale_acquirer_button button");
-        $pay_button.prop("disabled", false);
+odoo.define("website_sale_change_payment_fee.website_sale_fee", function(require) {
+    "use strict";
 
-        // When choosing a payment, update the quotation. Disable the 'Pay
-        // Now' button to avoid being redirected to payment acquirer if the delivery carrier update is
-        // not over.
-        var $payment_method = $("#payment_method");
-        var $selected_acquirer_id = $("input[name='selected_acquirer_id']");
-        var selected_acquirer_id = $selected_acquirer_id.val();
-        $payment_method.find("input[name='pm_id']").click(function(ev) {
-            var payment_id = $(ev.currentTarget)
-                .val()
+    require("web.dom_ready");
+    require("web.ajax");
+    require("web.core");
+
+    $(".o_payment_form").each(function() {
+        var payment_form = this;
+        var selected_acquirer = $("input[name='selected_acquirer_id']").val();
+
+        $(payment_form).on("click", 'input[name="pm_id"]', function() {
+            var clicked_acquirer = $(this)
+                .prop("value")
                 .substring(5);
-            // We need to check selected_acquirer_id because website_sale module clicks on checked acquirer at page load.
-            // See website_sale/static/src/js/website_sale_payment.js :
-            // .find("input[name='acquirer']:checked").click();
-            // Without this check, redirect would be performed right after page load
-            if (
-                selected_acquirer_id != null &&
-                payment_id != selected_acquirer_id
-            ) {
-                $pay_button.prop("disabled", true);
-                window.location.href = "/shop/payment?payment_fee_id=" + payment_id;
+            if (selected_acquirer !== clicked_acquirer) {
+                $("#o_payment_form_pay").addClass("disabled");
```

Comment:
Is this check needed at all? I didn't experience any strange behavior in my local environment disabling this condition.

If this is still needed, we should also revive the comments to make the intention clear.